website/docs/d/api_gateway_stage.html.markdown

```diff
@@ -0,0 +1,40 @@
+---
+layout: "aws"
+page_title: "AWS: aws_api_gateway_stage"
+sidebar_current: "docs-aws_api_gateway_stage"
+description: |-
+  Get information on an API Gateway Stage
+---
+
+# Data Source: aws_api_gateway_stage
+
+Use this data source to get the latest deploy id from a pre-existing stage. For example
+to update a stage's settings without overwriting the currently deployed stage.
+
+## Example Usage
+
+```hcl
+resource "aws_api_gateway_rest_api" "my_rest_api" {
+  name = "my-rest-api"
+}
+
+data "aws_api_gateway_stage" "production" {
+  rest_api_id   = "${aws_api_gateway_rest_api.test.id}"
+  stage_name = "production"
+}
+
+resource "aws_api_gateway_stage" "production" {
```

Comment:
Should this be named `prod` to match the `stage_name` below or vice versa? Maybe blue/green? Admittedly this example feels a little confusing to me for a use case since it doesn't show overwriting any stage settings. 😅

Reply:
Hopefully this is a bit better, our use case was:
We had a system that updated & deployed an api gateway stage, but wanted to configure settings (specifically the logging settings via terraform). This would cause issues every time we applied since a deployment is required as an attribute, this change will allow us to just use the latest deployment as input to the stage resource so we dont revery the api gateway on each trerraform apply!